lib/extensions/time.dart

```diff
@@ -0,0 +1,65 @@
+import 'package:flutter/material.dart';
+import 'package:sayilar/extensions/cases.dart';
+import 'package:sayilar/extensions/written_out.dart';
+
+/// An extension to get the written out name for a [TimeOfDay] *in turkish*.
+extension WrittenOut on TimeOfDay {
+  /// Get the full length written out name for `this` [TimeOfDay] *in turkish*.
+  String get writtenOut {
+    return switch (minute) {
+      0 => 'saat ${hour.writtenOut}',
+      15 => 'saat ${hour.writtenOut.accusative} çeyrek geçiyor',
+      < 30 => 'saat ${hour.writtenOut.accusative} ${minute.writtenOut} geçiyor',
+      30 => 'saat ${hour.writtenOut} buçuk',
+      // Alternative: 30 => 'Saat ${hours.writtenOut} dokuz',
+      45 => 'saat ${(hour + 1).writtenOut.dative} çeyrek var',
+      > 30 =>
+        'saat ${(hour + 1).writtenOut.dative} ${(60 - minute).writtenOut} var',
+      _ => throw UnimplementedError(
+          'This switch expression is logically exaustive, so this will never '
+          'be reached, but is needed to make the compiler happy.',
+        ),
+    };
+  }
+
+  List<String> get alternateWrittenOut {
+    return [
+      switch (minute) {
+        0 => hour.writtenOut,
+        15 => '${hour.writtenOut.accusative} çeyrek geçiyor',
+        < 30 => '${hour.writtenOut.accusative} ${minute.writtenOut} geçiyor',
+        30 => '${hour.writtenOut} buçuk',
+        // Alternative: 30 => 'Saat ${hours.writtenOut} dokuz',
+        45 => '${(hour + 1).writtenOut.dative} çeyrek var',
+        > 30 =>
+          '${(hour + 1).writtenOut.dative} ${(60 - minute).writtenOut} var',
+        _ => throw UnimplementedError(
+            'This switch expression is logically exaustive, so this will never '
+            'be reached, but is needed to make the compiler happy.',
+          ),
+      },
+    ];
+  }
+
+  /// Return a formatted string representing `this` [TimeOfDay].
+  String asString() => [
+        hour.toString(),
+        minute.toString().padLeft(2, '0'),
+      ].join(':');
```

Comment:
The default here should probably use `TimeOfDay.format` to get a format that matches the format in the time picker.
This can be added to the alternate strings though.

Reply:
Hard coded this behaviour now, which may not be perfect, but it should always use the 24 hour format, so it should be fine.